### DISCUSSION TITLE
confusing message of "Unmerged into origin/main"

### DISCUSSION THREAD

**Discussion by han1475** [magit/magit#4306] [Support] [answered]
hello,

In the `magit` buffer, `unstaged changes` means that I need to stage changes, `staged changes` means that I need to commit changes.

so I guess `Unmerged into origin/main` means that I need to merge commits, 
but it means that  I have commits in my local repository that hasn't been pushed in the remote repository.

I think this is a confusing message, it might be better to use `unpushed in the remote repository`.

Thanks a lot!

**Comment by tarsius** [comment, marked as answer]
If you have push access and are working directly on `main`, then yes, commits make it from `main` into `origin/main` by means of a *push*.

However if you are on `feature` then you get commits into `main` by checking out `main` and then *merging* `feature` into `main` (and then pushing *that* to `origin/main`. Likewise if you contribute to someone else's project, then you also don't push to the ultimate destination directly. First you push to `fork/feature` and open a pull-request, aka *merge-request*, and then they fetch, review and merge that into `main` before pushing that to `origin/main`.

**Reply by han1475** [reply]
Thanks for your reply, I get it.
